backend/pkg/utils/utils.go

```diff
@@ -10,3 +17,52 @@ func Contains[T comparable](elems []T, v T) bool {
 
 	return false
 }
+
+const (
+	// LevelInfo is the info level.
+	LevelInfo = iota
+	// LevelWarn is the warn level.
+	LevelWarn
+	// LevelError is the error level.
+	LevelError
+)
+
+// Log is a wrapper function for logging. It uses zlog package and logs to stdout.
+// It logs the message, source file and line number.
+// It logs the message at the level specified.
+func Log(level uint, str map[string]string, err interface{}, msg string) {
+	var event *zerolog.Event
+
+	switch level {
+	case LevelInfo:
+		event = zlog.Info()
+	case LevelWarn:
+		event = zlog.Warn()
+	case LevelError:
+		event = zlog.Error()
+	default:
+		event = zlog.Info()
+	}
+
+	for k, v := range str {
+		event.Str(k, v)
+	}
+
+	_, file, line, ok := runtime.Caller(1)
+	if ok {
+		event.Str("source", file)
+		event.Int("line", line)
+	}
+
+	if err != nil {
+		// Check if err is a error slice ([]error)
+		if errs, ok := err.([]error); ok {
+			event.Errs("error", errs).Msg(msg)
+		} else {
+			// Handle other error types
+			event.Err(err.(error)).Msg(msg)
+		}
```

Comment:
The code will reach this state even when a integer is passed as err. Either handle the state properly here or use custom type supported by Golang instead of interface in the function argument.